File: Spring_Intellij/healthifier/src/main/java/com/ricky/healthifier/service/tracker/FoodTrackerService.java
package com.ricky.healthifier.service.tracker;

import com.ricky.healthifier.datamodel.tracker.FoodTracker;
import com.ricky.healthifier.utils.exception.AppException;

import java.util.List;

public interface FoodTrackerService {

    boolean addFoodToTracker(FoodTracker foodTracker, String token) throws AppException;

    List<FoodTracker> getFoodTracker(String token) throws AppException;
}

File: Angular_VS/client/src/app/services/food-tracker/food-tracker.service.ts
import { FoodTracker } from './../../models/food-tracker/food-tracker';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TokenHttpService } from '../http/token-http.service';
import { Observable } from 'rxjs';
import * as Constants from '../../app.constants';

@Injectable({
  providedIn: 'root'
})
export class FoodTrackerService extends TokenHttpService {

  baseUrl = Constants.API + 'foodTrackers/';

  constructor(protected http: HttpClient) {
    super(http)
  }

  // Get All
  getFoodTracker(): Observable<any> {
    return this.get(this.baseUrl)
  }

  // Add
  addFoodToTracker(payload: FoodTracker): Observable<any> {
    return this.post(this.baseUrl, payload)
  }

  // Update
  updateFoodToTracker(id: number, payload: FoodTracker): Observable<any> {
    const url = this.baseUrl + id
    return this.put(url, payload)
  }

  // Delete
  deleteFoodFromTracker(id: number): Observable<any> {
    const url = this.baseUrl + id
    return this.delete(url)
  }
}
